hastexo/admin.py

```diff
@@ -99,7 +100,7 @@ class StackAdmin(admin.ModelAdmin):
     search_fields = ("name", "course_id", "status", "provider")
     readonly_fields = ("name", student_email, "course_id", "run", "protocol",
                        "port", "ip", "launch_timestamp", "suspend_timestamp",
-                       "created_on", "error_msg",)
+                       "created_on", "error_msg", "delete_age",)
```

Comment:
Help me understand this, please. If `delete_age` is read-only in the admin interface and `delete_by` is writable, what are the semantics of setting `delete_by` and leaving `delete_age` unmodified?

Reply:
Well, I saw two options for this:
1.` delete_age` get set each time the student starts or resumes their lab. It will look for the XBlock attribute value if that is `None`, use the global settings value. And with this approach, if the value is set in the admin it will simply be overridden when the student resumes, so having the option to edit it would be just pointless in this case. But this gives an option to edit the `delete_age` even mid-course for all students. The `delete_by` value can be set in the admin to optionally just throw away labs even before students resume (thinking that this would be a rare case).

2. `delete_age` gets set once when the student first launches the stack. It will look for the XBlock attribute value if that is `None`, use the global settings value. When the student resumes, if `delete_age` has a value, don't change it. So if we went and edited the value in the admin, the value will now persist. But we cannot make the change later via the XBlock attribute that would apply for all. If we want to change the delete_age for a single stack, we can do this here but if we would want to change it for all students, I am not sure how easy/comfortable would be doing that in the admin.

So I went with the first option and made the `delete_age` read-only, simply because the change would not take affect.